keras_hub/src/layers/modeling/rotary_embedding.py

```diff
@@ -69,33 +80,89 @@ def __init__(
         self,
         max_wavelength=10000,
         scaling_factor=1.0,
+        rope_type="linear",
+        beta_fast=32.0,
+        beta_slow=1.0,
+        original_max_position_embeddings=4096,
+        truncate=False,
         sequence_axis=1,
         feature_axis=-1,
         **kwargs,
     ):
         super().__init__(**kwargs)
         self.max_wavelength = max_wavelength
-        self.sequence_axis = sequence_axis
-        self.feature_axis = feature_axis
         self.scaling_factor = scaling_factor
-        self.built = True
+        self.rope_type = rope_type
+
+        # YaRN-specific parameters (only used when rope_type="yarn")
+        self.beta_fast = beta_fast
+        self.beta_slow = beta_slow
+        self.original_max_position_embeddings = original_max_position_embeddings
+        self.truncate = truncate
+
+        # Store original axis values for validation
+        self._original_sequence_axis = sequence_axis
+        self._original_feature_axis = feature_axis
+
```

Comment:
Change this back to         
self.sequence_axis = sequence_axis
self.feature_axis = feature_axis
 and add self.built = True